File: Programmers/maraton_loser.py
def solution(participant, completion):
    answer = ''
    p_dic = {}
    c_dic = {}
    p_set = set(participant)

    for p in participant:
        if p in p_dic:
            p_dic[p] += 1
        else:
            p_dic[p] = 1

    for c in completion:
        if c in c_dic:
            c_dic[c] += 1
        else:
            c_dic[c] = 1

    for p in p_set:
        if p in c_dic:
            p_dic[p] -= c_dic[p]

    for p in p_set:
        if p_dic[p] == 1:
            answer = p

    return answer

File: Programmers/Java/MaratonLoser.java
import java.util.HashMap;
import java.util.ArrayList;


public class MaratonLoser {
    public String solution(String[] participant, String[] completion) {
        String answer = "";

        HashMap<String, Integer> participant_map = new HashMap<>();
        HashMap<String, Integer> winners_map = new HashMap<>();

        for (String str : participant) {
            if (participant_map.containsKey(str)) {
                participant_map.put(str, participant_map.get(str) + 1);
            }
            else {
                participant_map.put(str, 1);
            }
        }

        for (String str : completion) {
            participant_map.put(str, participant_map.get(str) - 1);

        }

        ArrayList<String> keyList = new ArrayList<>(participant_map.keySet());
        for(String key : keyList) {
            if(participant_map.get(key) == 1) {
                answer = key;
            }
        }

        return answer;
    }
}
